turbinia/workers/file_system_timeline_test.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the FileSystemTimelineJob job."""
+
+from __future__ import unicode_literals
+
+import glob
+import unittest
+import os
+import mock
+
+from turbinia.evidence import BodyFile
+from turbinia.workers import file_system_timeline
+from turbinia.workers import TurbiniaTaskResult
+from turbinia.workers.workers_test import TestTurbiniaTaskBase
+
+from dfvfs.lib import definitions as dfvfs_definitions
+from dfvfs.path import factory as path_spec_factory
+
+
+class FileSystemTimelineTest(TestTurbiniaTaskBase):
+  """Tests for FileSystemTimelineJob."""
+
+  def setUp(self):
+    super(FileSystemTimelineTest, self).setUp(
+        task_class=file_system_timeline.FileSystemTimelineTask,
+        evidence_class=BodyFile)
+    self.setResults(mock_run=False)
+    self.task.output_dir = self.task.base_output_dir
+
+  def tearDown(self):
+    """Override tearDown method to deal with empty bodyfile."""
+    entries = glob.glob(self.task.output_dir + '/**', recursive=True)
```

Comment:
Doing file deletions like this always makes me a bit nervous just in the off chance that `self.task_output_dir` is empty (and so would delete '/').   I know the deletion pattern comes from the workers test, but there we are specifically adding individual files instead of globbing a full directory.   Could we potentially create a sub-directory which includes some kind of 'turbinia-test' or similar prefix that we can check for prior to deletion?  Could be similar to this: https://github.com/google/turbinia/blob/9269b925ec25342b4ac71f9bcfcd92660bf86e3b/turbinia/output_manager_test.py#L57

Reply:
Let me look into this a bit. The reason i created this function is because the test was fialing because the bodyfile is created by the task, but is not in the list that is used to delete files in the workers_test so rmdir() throws an exception as the directory is not empty. Maybe there is a way to get the file added to that list, or something similar to what you suggested.

Reply:
If it's only that one file, then you probably don't need to override this method at all, and in the test method you can just add the body file to `self.remove_files` and it will get automatically removed as part of the normal tear down.

Reply:
Fixed.